lib/rucio/tests/conftest.py

```
@@ -98,13 +105,38 @@ def mock_scope(vo):
     return InternalScope('mock', vo=vo)
 
 
+@pytest.fixture(scope='module')
+def test_scope(vo):
+    from rucio.common.types import InternalScope
+
+    return InternalScope('test', vo=vo)
+
+
 @pytest.fixture(scope='module')
 def root_account(vo):
     from rucio.common.types import InternalAccount
 
     return InternalAccount('root', vo=vo)
 
 
+@pytest.fixture(scope="module")
+def containerized_rses(rucio_client):
+    """
+    Detects if containerized rses for xrootd are available in the testing environment.
+    :return: A list of (rse_name, rse_id) tuples.
+    """
+    rses = []
+    try:
+        xrd_rses = [x['rse'] for x in rucio_client.list_rses(rse_expression='test_container_xrd=True')]
+        xrd_rses = [rucio_client.get_rse(rse) for rse in xrd_rses]
+        xrd_containerized_rses = [(rse_obj['rse'], rse_obj['id']) for rse_obj in xrd_rses if "xrd" in rse_obj['rse'].lower()]
+        xrd_containerized_rses.sort()
+        rses.extend(xrd_containerized_rses)
+    except Exception:
+        traceback.print_exc()
```

this seems like some failures will be silently ignored